src/blocks/custom.rs

```diff
@@ -97,6 +108,18 @@ impl Block for Custom {
 
         self.output.set_text(output);
 
+        if let Some(state_command) = self.state_command.clone() {
+            let state = Command::new(env::var("SHELL").unwrap_or("sh".to_owned()))
+                .args(&["-c", &state_command.clone()])
+                .output()
+                .map(|o| String::from_utf8_lossy(&o.stdout).trim().to_owned())
+                .unwrap_or_else(|e| e.description().to_owned())
```

Comment:
I'm not sure I understand what is happening here. My reading is: a UTF-8 parsing error will yield a string describing the error, which then generates an error in `try_into()` that suppresses this description, which then itself is suppressed by the `unwrap_or()`. If that is the case, what is the point of all the error handling?

Reply:
`from_utf8_lossy` will not return an error. Bytes that are not valid utf8 will just be replaced with �. The error conversion in unwrap_or_else is just so that we get back a string in any case, whether the command execution worked or not. We then take that string (which is either what the command returns, or an error message why that command failed) and try to create a state from that. If that fails, we assume "critical". We could save a bunch of code here with `map_or_else`, meaning we take the output and convert it to a state (and if that fails assume "critical") and in the same step convert errors from the command execution and map them to "critical".

Reply:
tbh, I'm a bit confused right now, [the docs](https://doc.rust-lang.org/std/result/enum.Result.html#method.map_or_else) describes map_or_else as a nightly only API with 2 parameters, and we use it with one parameter on stable...